types/keychain/index.d.ts

```diff
@@ -0,0 +1,43 @@
+// Type definitions for keychain 1.4
+// Project: https://github.com/drudge/node-keychain
+// Definitions by: Lucas Santos <https://github.com/khaosdoctor>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export interface KeyChainBaseOptions {
+    account: string;
+    service: string;
+    password: string;
+    type?: 'generic' | 'internet';
+}
+
+export enum KeychainErrorCode {
+    UnsupportedPlatform = 'UnsupportedPlatform',
+    NoAccountProvided = 'NoAccountProvided',
+    NoServiceProvided = 'NoServiceProvided',
+    NoPasswordProvided = 'NoPasswordProvided',
+    ServiceFailure = 'ServiceFailure',
+    PasswordNotFound = 'PasswordNotFound',
+}
+
+export type KeychainErrorType = `${KeychainErrorCode}Error`;
+
+export class KeychainError extends Error {
+    code: KeychainErrorCode;
+    type: KeychainErrorType;
+}
+
+declare const keychain: {
+    getPassword(
+        options: Pick<KeyChainBaseOptions, 'account' | 'service'>,
+        callback: (err: KeychainError, password: string) => void,
+    ): void;
+
+    setPassword(options: KeyChainBaseOptions, callback: (err: KeychainError) => void): void;
+
+    deletePassword(
+        options: Pick<KeyChainBaseOptions, 'account' | 'service'>,
+        callback: (err: KeychainError) => void,
+    ): void;
+};
+
+export default keychain;
```

Comment:
Thanks for clarifying! I had done this first but then got confused if it would be the best solution, then I removed everything and used the `export` syntax. But just got it back 😄

Reply:
also that way reflects an instance of class being exported:
```ts
// Type definitions for keychain 1.4
// Project: https://github.com/drudge/node-keychain
// Definitions by: Lucas Santos <https://github.com/khaosdoctor>
// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped

declare class KeychainAcces {
    getPassword(
        options: Pick<KeychainAcces.KeyChainBaseOptions, 'account' | 'service'>,
        callback: (err: KeychainAcces.KeychainError, password: string) => void,
    ): void;

    setPassword(
        options: KeychainAcces.KeyChainBaseOptions,
        callback: (err: KeychainAcces.KeychainError) => void,
    ): void;

    deletePassword(
        options: Pick<KeychainAcces.KeyChainBaseOptions, 'account' | 'service'>,
        callback: (err: KeychainAcces.KeychainError) => void,
    ): void;
}

declare namespace KeychainAcces {
    interface KeyChainBaseOptions {
        account: string;
        service: string;
        password: string;
        type?: 'generic' | 'internet';
    }

    type KeychainErrorCodes = 'UnsupportedPlatform'
        | 'NoAccountProvided'
        | 'NoServiceProvided'
        | 'NoPasswordProvided'
        | 'ServiceFailure'
        | 'PasswordNotFound';

    type KeychainErrorType = `${KeychainErrorCodes}Error`;

    class KeychainError extends Error {
        code: KeychainErrorCodes;
        type: KeychainErrorType;
    }
}

declare const keychain: KeychainAcces;

export = keychain;
```